### PR TITLE
Dev/594 cd overlays: How it supposes to be

### PR DESCRIPTION
**Pull request by a2313**

### CODE REVIEW

File: jdi-light-bootstrap-tests/src/main/java/io/github/com/sections/card/OverlaySection.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+public class OverlaySection extends Section {
+
+    @UI(".card-title")
+    public Text title;
+    @UI(".card-text-main")
+    public Text textMain;
+    @UI(".card-text-status")
+    public Text textStatus;
+}
```
Comment (mf-code) [inline jdi-light-bootstrap-tests/src/main/java/io/github/com/sections/card/OverlaySection.java]: no new line at the end of file

File: jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/card/CardImageOverlaysTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        baseValidation(cardImageOverlays.vectorImage);
+        baseValidation(cardImageOverlays.overlaySection.title);
+    }
+
+    @Test
+    public void classTest() {
+        cardImageOverlays.overlaySection.is().core().hasClass(IMAGE_TOP_CLASS);
+        cardImageOverlays.overlaySection.assertThat().core().hasClass(OVERLAY_CLASS);
+        cardImageOverlays.vectorImage.is().core().hasClass(IMAGE_TOP_CLASS);
+        //cardImageOverlays.vectorImage.assertThat().core().hasClass(IMAGE_TOP_CLASS);
```
Comment (mf-code) [inline jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/card/CardImageOverlaysTest.java]: should we remove it now?

File: jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/card/CardImageOverlaysTest.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    }
+}
```
Comment (mf-code) [inline jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/card/CardImageOverlaysTest.java]: no new line at the end of file

File: jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/asserts/BaseImageAssert.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    }
+    public BaseImageAssert height(int height) { return height(is(height)); }
+
+    @JDIAction("Assert that '{name}' width {0}")
+    public BaseImageAssert width(Matcher<Object> condition) {
+        jdiAssert(element.width(), condition);
+        return this;
+    }
+    public BaseImageAssert width(String width) { return width(is(width)); }
+}
```
Comment (mf-code) [inline jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/asserts/BaseImageAssert.java]: no new line at the end of file

File: jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/asserts/VectorImageAssert.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+public class VectorImageAssert extends BaseImageAssert<VectorImage> {
+
+    @JDIAction("Assert that '{name}' width {0}")
+    public VectorImageAssert width(Matcher<String> condition) {
+        jdiAssert(element.width(), condition);
+        return this;
+    }
+
+}
```
Comment (mf-code) [inline jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/asserts/VectorImageAssert.java]: now newline at the end of file

File: jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/elements/common/VectorImage.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
+package com.epam.jdi.light.ui.bootstrap.elements.common;
+
+import com.epam.jdi.light.elements.common.UIElement;
+import com.epam.jdi.light.ui.bootstrap.asserts.BaseImageAssert;
+import com.epam.jdi.light.ui.bootstrap.asserts.VectorImageAssert;
+import org.openqa.selenium.By;
+
+public class VectorImage extends BaseImage<VectorImageAssert> {
+
+    private UIElement getInnerElement(final String tagName) {
```
Comment (mf-code) [inline jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/elements/common/VectorImage.java]: why we dont put here @JDIAction ?

File: jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/card/CardImageOverlaysTest.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        bsPage.shouldBeOpened();
+    }
+
+    @Test
+    public void getTitleTest() {
+        assertEquals(cardImageOverlays.overlaySection.title.getText(), TITLE);
+    }
+
+    @DataProvider
+    private static Object[][] getElement() {
```
[the remark this answers is not in the document]
Comment (mf-code) [reply jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/card/CardImageOverlaysTest.java]: please fix codacy issue

[the remark this answers is not in the document]
Comment (a2313) [reply jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/card/CardImageOverlaysTest.java]: changed to public

File: jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/asserts/BaseImageAssert.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+import static org.hamcrest.Matchers.is;
+
+public abstract class BaseImageAssert<E extends BaseImage> extends UIAssert<BaseImageAssert, BaseImage> {
+
+    @JDIAction("Assert that '{name}' height {0}")
+    public BaseImageAssert height(Matcher<Integer> condition) {
+        jdiAssert(element.height(), condition);
+        return this;
+    }
+    public BaseImageAssert height(int height) { return height(is(height)); }
```
Comment (mf-code) [inline jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/asserts/BaseImageAssert.java]: JDIAction?

File: jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/asserts/BaseImageAssert.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+    @JDIAction("Assert that '{name}' width {0}")
+    public BaseImageAssert width(Matcher<Object> condition) {
+        jdiAssert(element.width(), condition);
+        return this;
+    }
+    public BaseImageAssert width(String width) { return width(is(width)); }
```
Comment (mf-code) [inline jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/asserts/BaseImageAssert.java]: JDIAction?

File: jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/asserts/BaseImageAssert.java (hunk 8 above)
Comment (a2313) [reply jdi-light-bootstrap/src/main/java/com/epam/jdi/light/ui/bootstrap/asserts/BaseImageAssert.java]: Could you add details to the note, please?

**Pull request merged**